### DISCUSSION TITLE
Async Wrapper

### DISCUSSION THREAD

**Discussion by SamirMishra27** [rogerbinns/apsw#456] [Q&A] [answered]
Hello, I would like to ask is there an **Async wrapper** of apsw, that can run SQLite operations asynchronously, without blocking the main thread?
I have a discord bot project which uses SQLite 3 to store its data, so I would need to run all database related operations asynchronously, because otherwise it would be blocking the whole process. 
That's pretty much it. I tried finding an async wrapper for this library but could not find anything.
Thanks!

**Comment by SamirMishra27** [comment]
Thank you for your reply! I will do what you recommended and use the `run_in_executor` for the processes that are taking unusually long, and let you know back if it worked! I am not sure how it will handle multiple SQLite operations at once, but since SQLite allows only one process to write at a time, I do not think I will have to worry about it.

I have not tried using APSW synchronously yet on my project. I will definitely give it a try. I used to use python's standard sqlite3 module synchronously, until the project grew bigger.

**Comment by rogerbinns** [comment, marked as answer]
Good news.  As of APSW 3.52.0.0, you get comprehensive [async support](https://rogerbinns.github.io/apsw/async.html) including all functionality and event loop implementations.
